### PR TITLE
Parameterize `hack/kind-up.sh`

### PR DESCRIPTION
**Pull request by briantopping**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
Parameterize the chart passed to `hack/kind-up.sh` such that the script can be used by extensions. This is necessary because vendored g/g does not include the `example` subtree. Access to that content means it must be duplicated into the tree of an extension that needs to use it, so it is no longer on a relative parent path to the `hack` directory.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by briantopping** [comment]
This is a minor modification of @rfranzke's recent work.

File: Makefile [hunk 1]
```diff
@@ -284,27 +284,27 @@ kind-ha-multi-zone-up kind-ha-multi-zone-down gardener-ha-multi-zone-up register
...
+	./hack/kind-up.sh  --chart $(REPO_ROOT)/example/gardener-local/kind/cluster --cluster-name gardener-local2 --environment $(KIND_ENV) --path-kubeconfig $(REPO_ROOT)/example/provider-local/seed-kind2/base/kubeconfig --path-cluster-values $(REPO_ROOT)/example/gardener-local/kind/local2/values.yaml --skip-registry
 kind2-down: $(KIND)
 	./hack/kind-down.sh --cluster-name gardener-local2 --path-kubeconfig $(REPO_ROOT)/example/provider-local/seed-kind2/base/kubeconfig --keep-backupbuckets-dir
 
 kind-ha-single-zone-up: $(KIND) $(KUBECTL) $(HELM)
-	./hack/kind-up.sh --cluster-name gardener-local-ha-single-zone --environment $(KIND_ENV) --path-kubeconfig $(REPO_ROOT)/example/provider-local/seed-kind-ha-single-zone/base/kubeconfig --path-cluster-values $(REPO_ROOT)/example/gardener-local/kind/ha-single-zone/values.yaml
+	./hack/kind-up.sh --chart $(REPO_ROOT)/example/gardener-local/kind/cluster --cluster-name gardener-local-ha-single-zone --environment $(KIND_ENV) --path-kubeconfig $(REPO_ROOT)/example/provider-local/seed-kind-ha-single-zone/base/kubeconfig --path-cluster-values $(REPO_ROOT)/example/gardener-local/kind/ha-single-zone/values.yaml
```
Comment (rfranzke) [inline Makefile]: Can we make this value the default in `kind-up.sh` so that we don't have to pass it here in every target?

Comment (briantopping) [reply Makefile]: Thanks for the feedback, I considered this and I guess it was late. Easy fix 👍 Thanks for all you do!

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Comment by timebertt** [comment]
Change as such is fine, but I would like to challenge the motivation behind it.
Before we start copying these charts to all extensions, can we think about a way to make them reusable? I would love to see a mechanism that allows me to run `make kind-up && make gardener-up` in all extensions for development :) 
We already reuse the `tools.mk` file and vendor several directories under `hack`. Can we use a similar mechanism for this?

/hold

**Comment by rfranzke** [comment]
@timebertt There is no need to hold this PR - it's a simple change which is self-contained and does not result in any technical debt. Making the charts reusable is not the goal of it and can be tackled separately.

**Comment by timebertt** [comment]
Right, holding the PR was too harsh. 
Just wanted add my two cents for some inspiration :) 

/hold cancel

**Comment by briantopping** [comment]
Thanks @timebertt i actually had exactly the same question for Rafael in a PM! He provided a trick to get it resolved and approved the PR. As I mentioned to him there, I was also agnostic by that point about retracting this PR.

**Pull request merged**